### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by ArmandoLiebenberg**
My submission for the COMP3710 report . Student ID is s4642427

This pull request contains the improved U-Net model for the ISIC datset segmentation task. I had some issues with loading the data and getting the predicted masks to plot correctly, but worked things out in the end. The average dice loss for the test set ended up being around 0.7, but the model was only trained for 20 epochs (Could potentially have trained for longer). 

 Refer to the README  for further details

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
**This is an initial inspection, no action is required at this point**
- Code: OK
- Results: 0.7 not idea but model is working;
- Readme: Results section formatting error "!!"; Code usage missing; References missing;
- Commit Messages: OK but not very consistently done over time.
- Other comments: For future reference, splits should not be a hyper-parameter for performance tuning

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem (low performance) -1
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage (no usage, refs etc.) -2
Module present
Commenting
No Data leakage
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe minimal, needs more info, background and results info for reproducibility -2
Good Description and Comments 
Markdown used PDF submitted (Markdown errors) -1
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description OK, could have more info -1
